test/schemes/fluid/surface_normal_sph.jl

```diff
@@ -0,0 +1,152 @@
+            @test true
+        end
+    end
+end
+
+@testset "Surface Normal Computation" begin
+    # Test case 1: Simple linear arrangement of particles
+    nparticles = 5
+    particle_spacing = 1.0
+    NDIMS = 2
+
+    coordinates = zeros(NDIMS, nparticles)
+    for i in 1:nparticles
+        coordinates[:, i] = [i * particle_spacing, 0.0]
+    end
+    velocity = zeros(NDIMS, nparticles)
+    mass = fill(1.0, nparticles)
+    fluid_density = 1000.0
+    density = fill(fluid_density, nparticles)
+
+    system, semi, ode = create_fluid_system(coordinates, velocity, mass, density,
+                                            particle_spacing;
+                                            buffer_size=0, NDIMS=NDIMS)
+
+    compute_and_test_surface_normals(system, semi, ode; NDIMS=NDIMS)
+end
+
+@testset "Sphere Surface Normals" begin
+    # Test case 2: Particles arranged in a circle
+    particle_spacing = 0.25
+    radius = 1.0
+    center = (0.0, 0.0)
+    NDIMS = 2
+
+    # Create a SphereShape (which is a circle in 2D)
+    sphere_ic = SphereShape(particle_spacing, radius, center, 1000.0)
+
+    coordinates = sphere_ic.coordinates
+    velocity = zeros(NDIMS, size(coordinates, 2))
+    mass = sphere_ic.mass
+    density = sphere_ic.density
+
+    # To get some what accurate normals we increase the smoothing length unrealistically
+    system, semi, ode = create_fluid_system(coordinates, velocity, mass, density,
+                                            particle_spacing;
+                                            buffer_size=0, NDIMS=NDIMS,
+                                            smoothing_length=3.0 * particle_spacing)
+
+    compute_and_test_surface_normals(system, semi, ode; NDIMS=NDIMS)
+
+    nparticles = size(coordinates, 2)
+    expected_normals = zeros(NDIMS, nparticles)
+    surface_particles = Int[]
+
+    # Compute expected normals and identify surface particles
+    for i in 1:nparticles
+        pos = coordinates[:, i]
+        r = pos - SVector(center...)
+        norm_r = norm(r)
+
+        if abs(norm_r - radius) < particle_spacing
+            expected_normals[:, i] = -r / norm_r
+
+            push!(surface_particles, i)
+        else
+            expected_normals[:, i] .= 0.0
+        end
+    end
+
+    # Normalize computed normals
+    computed_normals = copy(system.cache.surface_normal)
+    for i in surface_particles
+        norm_computed = norm(computed_normals[:, i])
+        if norm_computed > 0
+            computed_normals[:, i] /= norm_computed
+        end
+    end
+
+    # Compare computed normals to expected normals for surface particles
+    for i in surface_particles
+        @test isapprox(computed_normals[:, i], expected_normals[:, i], atol=0.05)
+    end
+
+    # Optionally, check that normals for interior particles are zero
+    # for i in setdiff(1:nparticles, surface_particles)
+    #     @test isapprox(norm(system.cache.surface_normal[:, i]), 0.0, atol=1e-4)
+    # end
```

Comment:
?

Reply:
This is also part of #584

Reply:
I thought I used it there but this is rather tricky... Even with the special conditions that are implemented in #584

Reply:
And with a larger tolerance? There should be some test to ensure the inner values are not completely random.
And there should not be a commented out test in here.

Reply:
The problem is that not only the outer layer has the surface normal set but this goes some layers deep so the only possiblity is to eliminate several of the outer layers.

Reply:
Than the inside is 0

Reply:
What happens between the surface and the inner layers where it's 0? Is this just a shorter normal then? Can you test that the normal either points in the correct direction or is almost zero?